server/test/handlers/getPossibleTimeslots.test.js

```diff
@@ -0,0 +1,140 @@
+const timeslotHandler = require('../../models/handlers/Timeslot');
+
+describe('testing getting possible timeslots functionality', () => {
+    beforeEach(() => {
+        jest.resetModules();
+    });
+
+    test('testing null parameters expecting full return of all records ', async () => {
+        // Arrange
+        const testTimeslotDetails = [
+            {
+                slot_id: 1,
+                start_time: '08:00:00',
+                end_time: '08:30:00'
+            },
+            {
+                slot_id: 2,
+                start_time: '08:30:00',
+                end_time: '09:00:00'
+            },
+            {
+                slot_id: 3,
+                start_time: '09:00:00',
+                end_time: '09:30:00'
+            },
+            {
+                slot_id: 4,
+                start_time: '09:30:00',
+                end_time: '10:00:00'
+            },
+            {
+                slot_id: 5,
+                start_time: '10:00:00',
+                end_time: '10:30:00'
+            },
+            {
+                slot_id: 6,
+                start_time: '10:30:00',
+                end_time: '11:00:00'
+            },
+            {
+                slot_id: 7,
+                start_time: '11:00:00',
+                end_time: '11:30:00'
+            },
+            {
+                slot_id: 8,
+                start_time: '11:30:00',
+                end_time: '12:00:00'
+            },
+            {
+                slot_id: 9,
+                start_time: '12:00:00',
+                end_time: '12:30:00'
+            },
+            {
+                slot_id: 10,
+                start_time: '12:30:00',
+                end_time: '13:00:00'
+            },
+            {
+                slot_id: 11,
+                start_time: '13:00:00',
+                end_time: '13:30:00'
+            },
+            {
+                slot_id: 12,
+                start_time: '13:30:00',
+                end_time: '14:00:00'
+            },
+            {
+                slot_id: 13,
+                start_time: '14:00:00',
+                end_time: '14:30:00'
+            },
+            {
+                slot_id: 14,
+                start_time: '14:30:00',
+                end_time: '15:00:00'
+            },
+            {
+                slot_id: 15,
+                start_time: '15:00:00',
+                end_time: '15:30:00'
+            },
+        ];
+
+        // Act
+        const possibleTimeslots = await timeslotHandler.getPossibleTimeslots();
+
+        // Assert
+        expect(possibleTimeslots).toMatchObject(testTimeslotDetails);
+    });
+
+
+    test('testing specified time parameters expecting a return', async () => {
```

Comment:
Hi Peter,
Another possible unit test may be invalid date parameters, i.e., 'hello' or true is specified for the `startTime` and consider how the method should respond.

Reply:
Hi amy, I require assistance on this issue as your comment above and this one touch on the same error checking topic. Unsure of how to properly implement error checking while allowing null values to be passed for startTime and endTime

Reply:
Hi Peter, 
I see. I have made a new comment on the specific line where the error checking may be implemented in the `/db/timeslot.js` file. Feel free to check out my suggestion and let me know if it is unclear.

Reply:
Comment thread relevant to part of PR no longer in development.